Source/WebCore/html/CanvasBase.cpp

```diff
@@ -361,12 +368,50 @@ RefPtr<ImageBuffer> CanvasBase::allocateImageBuffer(bool usesDisplayListDrawing,
 
 bool CanvasBase::shouldInjectNoiseBeforeReadback() const
 {
+    // Note, every early-return resulting from this check potentially leaks this state. This is a risk that we're accepting right now.
     return scriptExecutionContext() && scriptExecutionContext()->noiseInjectionHashSalt();
 }
 
-bool CanvasBase::postProcessPixelBuffer(Ref<PixelBuffer>&& pixelBuffer, bool wasLastDrawByBitMap, const HashSet<uint32_t>& suppliedColors) const
+IntRect CanvasBase::computeImageDataRect(const ImageBuffer& buffer, int width, int height, IntRect destRect, const IntSize& destOffset)
```

Comment:
This function is relevant only when computing the source rect used in getImageData() Context2D call. It's an odd function used for odd reason (and implemented in a bit odd way), so it should stay at its odd place, at Context2D.



Here in CanvasBase we are concerned about the imagebuffer, and there is nothing mysterious about its sizes.

Reply:
I factored out that code in 1a169d095a6691b2461810e3b639fffed1cf87f6, so we can restore the previous code. I'll do that in a follow-up patch.